src/modules/requirements/TimeRequirement.ts

```diff
@@ -1,33 +1,26 @@
 import { AchievementOption } from '../GameConstants';
 import GameHelper from '../GameHelper';
 import Requirement from './Requirement';
+import DayCyclePart from '../dayCycle/DayCyclePart';
+import DayCycle from '../dayCycle/DayCycle';
 
 export default class TimeRequirement extends Requirement {
     private updateTrigger = ko.observable(0);
-    constructor(public startHour: number, public endHour: number, option = AchievementOption.more) {
+    constructor(public dayCycleParts: DayCyclePart[], option = AchievementOption.more) {
         super(1, option);
         setInterval(
             () => GameHelper.incrementObservable(this.updateTrigger),
             60 * 1000,
         );
     }
 
-    public getProgress() {
+    public getProgress(): number {
         this.updateTrigger();
 
-        const [startHour, endHour] = [this.startHour, this.endHour];
-        const currentHour = new Date().getHours();
-
-        const satisfied = startHour < endHour
-            // If the start time is before the end time, both need to be true
-            ? currentHour >= startHour && currentHour < endHour
-            // If the start time is after the end time, only 1 needs to be true
-            : currentHour >= startHour || currentHour < endHour;
-
-        return Number(satisfied);
+        return Number(this.dayCycleParts.includes(DayCycle.currentDayCyclePart()));
     }
 
     public hint(): string {
-        return `Your local time must be between ${this.startHour}:00 and ${this.endHour}:00`;
+        return `Your local part of the day must be ${this.dayCycleParts.map((dayCyclePart) => DayCyclePart[dayCyclePart]).join(' or ')}`;
```

Comment:
This feels less informative, unless there is somewhere telling people exactly what time the dayCycleParts start/end?
If there isn't, can we do that somewhere, or change this hint to give the actual times (either in addition or instead of the day part name)

Reply:
There is a display of the part of the day now, but that doesn't allow knowing when the part starts. Maybe should I add an NPC? that describes their day : "i wake up at dawn, at 6 AM every day" and so on ?